source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -646,8 +646,11 @@ ShaderGraphPtr ShaderGraph::create(const ShaderGraph* parent, const string& name
             // Handle node input ports
             for (const ValueElementPtr& nodedefPort : nodeDef->getActiveValueElements())
             {
-                if (nodedefPort->isA<Output>())
+                InputPtr nodedefInputPort = nodedefPort->asA<Input>();
```

Comment:
Only consider inputs as tokens are also value elements and this would try and set values on tokens and fail.